docs/current/sailbot_workspace/run.md

```diff
@@ -1,34 +1,94 @@
 
-## 3. Run the ROS program
+    One way to interface with Git is through CLI commands. However, you may find it faster to use
+    [VS Code's interface](https://code.visualstudio.com/docs/sourcecontrol/overview){target=_blank},
+    especially when working with multiple repositories.
 
-Run a ROS launch file or package-specific executable with `ros2 launch ...` or `ros2 run ...`, respectively.
+Things to note when making changes:
 
-??? tip "Running GUI applications on macOS"
+- When C++ or Python files are saved, you may notice that some lines change. We use formatters to help fix lint errors;
+  not all lint errors can be fixed by formatters, so you may have to resolve some manually
+- When changing a package's source files, you likely should update its test files accordingly
 
-    If you want to run GUI applications on macOS, ensure that XQuartz is running.
+## 3. Build your changes
+
+!!! info ""
+
+    Revamped in [:octicons-tag-24: v1.2.0](https://github.com/UBCSailbot/sailbot_workspace/releases/tag/v1.2.0){target=_blank}
+
+In general, changes need to be built before they can be run. You can skip this step if you only modified Python source
+or test files (in `python_package/python_package/` or `python_package/test`, respectively), or are running a launch type
+launch configueration.
+
+1. Run the `Build` VS Code task: ++ctrl+shift+b++
+    - On macOS the shortcut is ++cmd+shift+b++
+2. Depending on which packages you modified, select whether to build all packages or a single one
+3. Unless you want to run `clang-tidy`, use the `-q` build argument (default) for quicker build times
+
+## 4. Verify your changes
+
+!!! info ""
+
+    Revamped in [:octicons-tag-24: v1.2.0](https://github.com/UBCSailbot/sailbot_workspace/releases/tag/v1.2.0){target=_blank}
+
+1. Run the package(s) that your changes are in: the run commands for each package should be documented in their READMEs,
+   but in general they can be run using a VS Code or CLI command:
+    - VS Code:
+        - Launch files: `ROS: Run a ROS launch file (roslaunch)`
+        - Nodes: `ROS: Run a ROS executable (rosrun)`
+    - CLI:
+        - Launch files: `ros2 launch <package> <launch file>`
+        - Nodes: `ros2 run <package> <executable>`
+
+    ??? tip "Running GUI applications on macOS"
+
+        If you want to run GUI applications on macOS, ensure that XQuartz is running.
+
+2. Debug your changes if they aren't behaving how you expect by setting breakpoints and running one of our launch
+   configurations in the "Run and Debug" sidebar panel; launch configuration types:
+    - Launch: runs the desired launch file or executable
+        - For launch files, `ROS: Launch`
+        - For C++ executables, `C++ (GDB): Launch`
+    - Attach: attaches to a running executable
+        - `ROS: Attach`
+3. Run lint and test tasks to make sure you changes will pass our CI:
+    - `ament lint`
+    - For C++ packages, `clang-tidy`
+    - `test`
 
 ## Troubleshooting
```

Comment:
https://github.com/UBCSailbot/docs/blob/c419b25dbe13b3557f9b472ae43f5f892ddc12b2/docs/current/sailbot_workspace/run.md?plain=1#L93-L96

This isn't for "Raye's" software necessarily.